src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -627,10 +615,28 @@ void CascadiaSettings::LayerJson(const Json::Value& json)
 void CascadiaSettings::_LayerOrCreateProfile(const Json::Value& profileJson)
 {
     // Layer the json on top of an existing profile, if we have one:
-    auto pProfile = _FindMatchingProfile(profileJson);
-    if (pProfile)
+    auto profileIndex{ _FindMatchingProfileIndex(profileJson) };
+    if (profileIndex)
     {
-        pProfile->LayerJson(profileJson);
+        auto parentProj{ _profiles.GetAt(*profileIndex) };
+        auto parent{ winrt::get_self<Profile>(parentProj) };
+
+        if (_userDefaultProfileSettings)

```

Comment:
The Profile Inheritance Graph should look like this (layer settingsA onto settingsB):
```
[A]          [B]          [prof.def]
bar.exe ---> foo.exe ---> cmd.exe 
                     \--> null Profile
```

But because we hold one reference to `_userDefaultProfileSettings`, you're right, we end up doing this:
```
[A]          [prof.def]   
bar.exe ---> cmd.exe
        \--> foo.exe [B] ---> cmd.exe [prof.def] 
                         \--> null Profile
```

So I think we may need to make `CascadiaSettings` inheritable to get around that issue? But yeah, this'll only really be a problem when we allow multiple files being layered (settings.json and defaults.json don't count because there's special handling for defaults.json; we know there isn't a profiles.defaults in defaults.json).